src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -333,80 +356,143 @@ export const isMyInfoCookie = (
  * @return err(MyInfoCookieStateError) if the extracted myInfoCookie was in an error state
  * @return err(MyInfoCookieAccessError) if the cookie has been accessed before
- */
-export const extractAndAssertMyInfoCookieValidity = (
+  */
+export const extractAndAssertOldMyInfoCookieValidity = (
   cookies: Record<string, unknown>,
 ): Result<
-  MyInfoSuccessfulCookiePayload,
-  | MyInfoCookieStateError
-  | MyInfoMissingAccessTokenError
-  | MyInfoCookieAccessError
+  MyInfoSuccessfulOldCookiePayload,
+  MyInfoCookieStateError | MyInfoMissingAccessTokenError
 > =>
-  extractMyInfoCookie(cookies)
-    .andThen((cookiePayload) => assertMyInfoCookieSuccessState(cookiePayload))
-    .andThen((cookiePayload) => assertMyInfoCookieUnused(cookiePayload))
-
-/**
- * Asserts that the myInfoCookie has not been used before
- * @param myInfoCookie
- * @returns ok(myInfoCookie) if the cookie has not been used before
- * @returns err(MyInfoCookieAccessError) if the cookie has been used before
- */
-export const assertMyInfoCookieUnused = (
-  myInfoCookie: MyInfoSuccessfulCookiePayload,
-): Result<MyInfoSuccessfulCookiePayload, MyInfoCookieAccessError> =>
-  myInfoCookie.usedCount <= 0
-    ? ok(myInfoCookie)
-    : err(new MyInfoCookieAccessError())
+  extractOldMyInfoCookie(cookies).andThen((cookiePayload) =>
+    assertOldMyInfoCookieSuccessState(cookiePayload),
+  )
 
 /**
+ * TODO(#5452): Delete this function
  * Extracts access token from a MyInfo cookie
  * @param cookie Cookie from which access token should be extracted
  */
-export const extractAccessTokenFromCookie = (
-  cookie: MyInfoCookiePayload,
+export const extractAccessTokenFromOldCookie = (
+  cookie: MyInfoOldCookiePayload,
 ): Result<string, MyInfoCookieStateError> => {
-  if (cookie.state !== MyInfoCookieState.Success) {
+  if (cookie.state !== MyInfoOldCookieState.Success) {
     return err(new MyInfoCookieStateError())
   }
   return ok(cookie.accessToken)
 }
 
+/**
+ * Extracts a MyInfo login cookie from a request's cookies
+ * @param cookies Cookies in a request
+ */
+export const extractMyInfoLoginJwt = (
+  cookies: Record<string, unknown>,
+): Result<string, MyInfoMissingLoginCookieError> => {
+  const jwt = cookies[MYINFO_LOGIN_COOKIE_NAME]
+  if (typeof jwt === 'string' && !!jwt) {
+    return ok(jwt)
+  }
+  return err(new MyInfoMissingLoginCookieError())
+}
+
+/**
+ * Extracts a MyInfo auth code cookie from a request's cookies, and validates
+ * its shape.
+ * @param cookies Cookies in a request
+ */
+export const extractAuthCode = (
+  cookie: unknown,
+): Result<
+  string,
+  MyInfoInvalidAuthCodeCookieError | MyInfoCookieStateError
+> => {
+  if (!isMyInfoAuthCodeCookie(cookie)) {
+    return err(new MyInfoInvalidAuthCodeCookieError(cookie))
+  }
+  if (cookie.state !== MyInfoAuthCodeCookieState.Success) {
+    return err(new MyInfoCookieStateError())
+  }
+  return ok(cookie.authCode)
```

Comment:
should we add in a check that authCode is non-empty string? there is no guarantee from the type, and we use this directly to retrieve the access token in L213 of `myinfo.service`